service/module.go

```diff
@@ -48,57 +47,68 @@ type Module interface {
 
 // ModuleProviderFromFunc creates a new ModuleProvider from a name and create function.
 func ModuleProviderFromFunc(name string, createFunc func(Host) (Module, error)) ModuleProvider {
-	return &moduleProviderFunc{name, createFunc}
+	return &moduleProvider{name: name, createFunc: createFunc}
 }
 
 // ModuleOption is a function that configures module creation.
 type ModuleOption func(*moduleOptions) error
 
-// WithModuleName will override the name given by the ModuleProvider.
-func WithModuleName(name string) ModuleOption {
+// WithName will override the root service name specified in config.
+func WithName(name string) ModuleOption {
 	return func(o *moduleOptions) error {
-		o.name = name
+		o.ServiceName = name
 		return nil
 	}
 }
 
-// WithModuleRole will add a role to the Module.
+// WithRole will add a role to the Module.
 //
 // If the role was already added, this will be a no-op.
-func WithModuleRole(role string) ModuleOption {
+func WithRole(role string) ModuleOption {
 	return func(o *moduleOptions) error {
 		if role == "" {
 			return nil
 		}
-		for _, elem := range o.roles {
+		for _, elem := range o.Roles {
 			if role == elem {
 				return nil
 			}
 		}
-		o.roles = append(o.roles, role)
+		o.Roles = append(o.Roles, role)
 		return nil
 	}
 }
 
+// WithModuleName will override the name given by the ModuleProvider.
+func WithModuleName(name string) ModuleOption {
+	return func(o *moduleOptions) error {
+		o.ModuleName = name
+		return nil
+	}
+}
+
+// moduleOptions specifies options for service name and role
+type moduleOptions struct {
+	ModuleName  string
+	ServiceName string
+	Roles       []string
+}
+
 // NewScopedHost returns a new Host scoped to a module. This should generally be used for testing.
-func NewScopedHost(host Host, name string, roles ...string) (Host, error) {
-	return newScopedHost(host, name, roles...)
+func NewScopedHost(host Host, moduleName, serviceName string, roles ...string) (Host, error) {
+	return newScopedHost(host, moduleName, serviceName, roles...), nil
 }
 
-type moduleProviderFunc struct {
+type moduleProvider struct {
 	name       string
 	createFunc func(Host) (Module, error)
 }
 
-func (m *moduleProviderFunc) DefaultName() string              { return m.name }
-func (m *moduleProviderFunc) Create(host Host) (Module, error) { return m.createFunc(host) }
-
-type moduleOptions struct {
-	name  string
-	roles []string
-}
+func (m *moduleProvider) DefaultName() string              { return m.name }
+func (m *moduleProvider) Create(host Host) (Module, error) { return m.createFunc(host) }
 
 type moduleWrapper struct {
+	name       string
```

Comment:
this is moduleName, I did not explicitly call it so because the struct itself says moduleWrapper.